torch/distributed/tensor/parallel/style.py

```diff
@@ -299,32 +304,6 @@ def _get_prepare_input(
     Get the prepare input function for this parallel style.
     """
 
-    def _redistribute_per_both_layouts(t, input_layout, output_layout, device_mesh):
-        dst_placements = (output_layout,)
-        if isinstance(t, DTensor):
-            return (
-                t
-                if _needs_redistribute(dst_placements, t)
-                else t.redistribute(device_mesh, dst_placements)
-            )
-        elif isinstance(t, torch.Tensor):
-            dtensor = DTensor.from_local(
-                t, device_mesh, [input_layout], run_check=False
-            )
-            return (
-                dtensor
-                if _needs_redistribute(dst_placements, dtensor)
-                else dtensor.redistribute(device_mesh, dst_placements)
-            )
-        else:
-            if input_layout is not None:
-                raise RuntimeError(
-                    "Tensor parallel module expects DTensor or tensor"
-                    f" when layout specified but received {type(t)}!"
-                )
-            else:
-                return t
-
     def make_input_redistribute_1d(
```

Comment:
Do you really need to separate out `make_input_redistribute_1d` and `make_out_redistribute_1d`? They are doing the same thing, we should:
1. just use one function instead of using two separate path
2. probably delete `_get_prepare_input` and `_get_prepare_output` and just put that functools.partial thing inside the PrepareModuleInput/Output itself for simplicity

Reply:
yes, will moving inside.